alphacsc/init_dict.py

```diff
@@ -132,47 +132,38 @@ def wrap_rank1(self, D_hat):
 
 class BaseStrategy():
 
-    def __init__(self, generator):
-        self.generator = generator
-        self.n_channels = generator.n_channels
-        self.n_atoms = generator.n_atoms
-        self.n_times_atom = generator.n_times_atom
-        self.rng = generator.rng
-
-    def get_D_shape(self):
-        return self.generator.get_D_shape()
+    def __init__(self, parent):
+        self.parent = parent
+        self.n_channels = parent.n_channels
+        self.n_atoms = parent.n_atoms
+        self.n_times_atom = parent.n_times_atom
+        self.rng = parent.rng
 
-    def get_dict(self, X, D_init_params):
+    def init(self, X, D_init_params):
         raise NotImplementedError()
```

Comment:
Not sure it is worth having this extra abstraction?
Maybe better to simply have functions that returns  `D_init` given `cdl_params` containing what is contained in `init`?